### PR TITLE
Passkey Signature Pallet Design Doc

### PR DESCRIPTION
**Pull request by saraswatpuneet**
# Goal
The goal of this PR is 

Closes 

# Discussion

# Checklist
- [ ] Chain spec updated
- [ ] Custom RPC OR Runtime API added/changed? Updated js/api-augment.
- [x] Design doc(s) updated
- [ ] Tests added
- [ ] Benchmarks added
- [ ] Weights updated

### CODE REVIEW

File: designdocs/passkey_p256.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+## 1. Introduction
+
+This document outlines the design considerations and specifications for integrating P256 Passkey support for performing transactions on Frequency chain. Passkey support aims to provide a novel non custodial solution for managing user accounts and signing transactions on-chain.
+
+## 2. Terminology
+
+### Accounts
+
+- **Authorized Account**: The user account associated with the Passkey.
+- **Parent Account**: The account holding the primary private key for transaction signing.
```
Comment (aramikm) [inline designdocs/passkey_p256.md:36]: What s the different from Authorized account and parent account? I think using `parent` is confusing in here since there no hierarchy like that.

File: designdocs/passkey_p256.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+- **Frequency Access**: The platform facilitating the management and interaction with user accounts.
+
+### Keys
+
+- **Passkey**: P256 key pair used for transaction signing and account management. This is the primary key used for transaction signing.
+- **Passkey Public Key**: The public key derived from the Passkey private key.
+- **Seed Phrase**: A mnemonic phrase used to generate cryptographic keys, particularly for SR25519 accounts. This is used for signing passkey proving account ownership.
```
Comment (aramikm) [inline designdocs/passkey_p256.md:43]: I recommend replace seed phrase with Frequency key so we would have 2 keys
1- Passkey
2- Frequency Account Key

Comment (aramikm) [reply designdocs/passkey_p256.md:43]: We might need some abbreviation for both, to make it easier

File: designdocs/passkey_p256.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+- **Seed Public Key**: The public key derived from the Seed Phrase.
+
+### Signatures
+
+- **Seed signature on PassKey Public Key**: A cryptographic signature generated using the Seed Phrase private key. The data being signed is public key derived from passkey. This is used to prove ownership of parent account.
```
Comment (aramikm) [inline designdocs/passkey_p256.md:48]: Frequency Account key signature of PassKey public key:

having something like `freq_keypair.sign(pass_key.public_key)` would also be helpful

File: designdocs/passkey_p256.md (hunk 1 above)
Comment (saraswatpuneet) [reply designdocs/passkey_p256.md:36]: I renamed them to `Passkey Account` and `Frequency Account` where one is passkey and other is used for execution txs

File: designdocs/passkey_p256.md (hunk 2 above)
Comment (saraswatpuneet) [reply designdocs/passkey_p256.md]: For now i will call them `Account Keys` or `Account Public Key` so its generic

**Comment by aramikm** [comment]
Submitted diagram for Registration and Transaction flows:

UC3 0_ User Submits Signed with Passkeys Transaction (Sequence Diagram)
UC1 0_ User Registers New Frequency Access Account with Passkeys (Sequence Diagram)

**Comment by aramikm** [comment]
Simplified diagrams

Passkey-Passkey Registration

Passkey-Passkey Transaction

File: designdocs/passkey_p256.md [hunk 4]
```diff
@@ -0,0 +1,552 @@
...
+### Terms
+
+These are the following technical terms and specifications for the implementation of Passkey support on Frequency chain.
+
+- `account_keypair` : A cryptographic key pair generated from a seed phrase
+- `account_pk` : Account Public Key
+- `passkey_pk` : Passkey Public Key
+- `passkey_sig_pk` : Signature generated by Passkey on  `account_pk`
+- `account_sig_passkey_pk` : Signature generated by Account Key on `passkey_pk`
+- `passkey_tx_signature` : Signature generated by Passkey on transaction payload, this is required for transaction submission.
```
Comment (shannonwells) [inline designdocs/passkey_p256.md:123]: why the change of naming pattern? Is there some reason it should not be `passkey_sig_tx[n]`?

File: designdocs/passkey_p256.md [hunk 5]
```diff
@@ -0,0 +1,552 @@
...
+## 5. Specification
+
+### Terms
+
+These are the following technical terms and specifications for the implementation of Passkey support on Frequency chain.
+
+- `account_keypair` : A cryptographic key pair generated from a seed phrase
+- `account_pk` : Account Public Key
+- `passkey_pk` : Passkey Public Key
+- `passkey_sig_pk` : Signature generated by Passkey on  `account_pk`
```
Comment (shannonwells) [inline designdocs/passkey_p256.md:121]: If these keys are signing each other, I think they need to be named in similar ways, e.g. `passkey_sig_account_pk` and `account_sig_passkey_pk`

File: designdocs/passkey_p256.md [hunk 6]
```diff
@@ -0,0 +1,552 @@
...
+- Choose vetted libraries, npm packages for handling cryptographic operations, such that signing with seed phrase, or handling of private keys is done securely.
+
+#### Front-end (client)
+
+- If key generation is done in front-end, it should ideally being done inside an isolated section such as iframe or Web Worker.
+- Generated Keypair should not get stored permanently (except for back up options) and removed as soon as it is not required.
+
+#### Backend
+
+- Passkey registration response should get verified which checks the random challenge
```
Comment (shannonwells) [inline designdocs/passkey_p256.md:312]: This is the first time a random challenge is mentioned.  What is the random challenge here?

**Review by shannonwells — COMMENTED** [review]
If I understand correctly, in summary, this process generates two keypairs in one session, which sign each other.  One is the Passkey Pair and the other is the Account Pair which is the on chain account key. This allows any Passkey signature for Account key operations to be guaranteed to be authorized, because the two first proofs (the keys signing each other) - which are stored by WebWallet - prove 
1. That I own the Passkey (obviously)
2. That I also own the Account key because I, the Passkey holder, was present when its seed phrase was generated.

Is that right?  Even if it's wrong on some details, I think some type of summary of the process would be useful at the beginning.

File: designdocs/passkey_p256.md (hunk 4 above)
Comment (aramikm) [reply designdocs/passkey_p256.md:123]: correct, we have not been consistent

File: designdocs/passkey_p256.md (hunk 6 above)
Comment (aramikm) [reply designdocs/passkey_p256.md]: Added details

File: designdocs/passkey_p256.md [hunk 7]
```diff
@@ -0,0 +1,556 @@
...
+        - [Pros/Cons](#proscons-1)
+      - [Extending MultiSignature](#extending-multisignature)
+        - [Pros/Cons](#proscons-2)
+    - [Generic Key Support](#generic-key-support)
+    - [Separate Pallet](#separate-pallet)
+  - [8. References](#8-references)
+
+## 1. Introduction
+
+This document outlines the design considerations and specifications for integrating P256 Passkey support for performing transactions on Frequency chain. Passkey support aims to provide a novel non custodial solution for managing user accounts and signing transactions on-chain.
```
Comment (JoeCap08055) [inline designdocs/passkey_p256.md:48]:
```suggestion
This document outlines the design considerations and specifications for integrating P256 Passkey support for performing transactions on the Frequency chain. Passkey support aims to provide a novel non-custodial solution for managing user accounts and signing transactions on-chain.
```

File: designdocs/passkey_p256.md [hunk 8]
```diff
@@ -0,0 +1,556 @@
...
+
+## 2. Benefits
+
+- **Non-Custodial Account Management**: Passkey support enables users to interact with Frequency chain without actual seed access. This reduces the risk of seed exposure and enhances security.
```
Comment (JoeCap08055) [inline designdocs/passkey_p256.md:52]:
```suggestion
- **Non-Custodial Account Management**: Passkey support enables users to interact with the Frequency chain without actual seed access. This reduces the risk of seed exposure and enhances security.
```

File: designdocs/passkey_p256.md [hunk 9]
```diff
@@ -0,0 +1,556 @@
...
+
+- **Functionality**: Passkey support provides a better user experience by enabling users to sign transactions without seed access. The poliferation of passkey enabled services can lead to a seamless user experience.
```
Comment (JoeCap08055) [inline designdocs/passkey_p256.md:54]:
```suggestion
- **Functionality**: Passkey support provides a better user experience by enabling users to sign transactions without seed access. The proliferation of passkey-enabled services can lead to a seamless user experience.
```

File: designdocs/passkey_p256.md [hunk 10]
```diff
@@ -0,0 +1,556 @@
...
+
+- **Interoperability**: The use of standard cryptographic protocols (e.g., P256) ensures compatibility with other systems and services that also support these standards. This can facilitate integrations with existing wallets, security modules, and third-party applications.
+
+- **Backup and Recovery**: Passkey support can be used to facilitate account recovery and backup mechanisms. By signing the account public key, users can prove ownership of the account and recover access in case of loss of seed access.
```
Comment (JoeCap08055) [inline designdocs/passkey_p256.md:58]: Perhaps:

```suggestion
- **Backup and Recovery**: Passkey support can be used to facilitate account recovery and backup mechanisms. By signing the account public key, users can prove ownership of the account and recover access in case of a lost seed.
```

File: designdocs/passkey_p256.md [hunk 11]
```diff
@@ -0,0 +1,556 @@
...
+### Keys
+
+- **Passkey**: P256 key pair used for transaction signing and account management. This is the primary key used for transaction signing.
+- **Passkey Public Key**: The public key retrieved from the Passkey registration process.
+- **Account Key**: The key pair generated from a seed phrase.
+- **Account Public Key**: The public key derived from the Account Key.
+
+### Signatures
+
+- **Account signature on PassKey Public Key**: A cryptographic signature generated using the Account Key. The data being signed is passkey public key. This is used to prove ownership of Account Key.
```
Comment (JoeCap08055) [inline designdocs/passkey_p256.md:77]:
```suggestion
- **Account signature on PassKey Public Key**: A cryptographic signature generated using the Account Key. The data being signed is the Passkey Public Key. This is used to prove ownership of Account Key.
```

File: designdocs/passkey_p256.md [hunk 12]
```diff
@@ -0,0 +1,556 @@
...
+- Frequency Account key signature of Passkey public key (`account_sig_passkey_pk`)
+- Generated Account public key (`account_pk`)
+- `credentialPublicKey`: Passkey Public key (`passkey_pk`)
+- `UserName`: User's website-specific username (email, etc...)
+- `UserId`: User's website-specific unique ID
+- `CredentialId`: The credential's credential ID for the pass key
+- `Counter`: The number of times the authenticator reported it has been used.
+- `credentialDeviceType`: Whether this is a single-device or multi-device credential.
+- `credentialBackedUp`: Whether or not the multi-device credential has been backed up. Always false for single-device credentials.
+- Transaction call
```
Comment (JoeCap08055) [inline designdocs/passkey_p256.md:105]: General comment: perhaps make a pass for consistency of capitalization, ie 'PassKey' vs 'Pass key' vs 'passkey', or 'Public Key' vs 'Public key' vs 'public key', etc...

File: designdocs/passkey_p256.md [hunk 13]
```diff
@@ -0,0 +1,556 @@
...
+        const registerPassKey = await SimpleWebAuthnBrowser.startRegistration(options);
+        ...
+        ...
+        // Sign Passkey Public Key using Account Key
+        const passkey_sig_account_pk = registerPassKey.signature; // obtained from the registration response
+        const passkey_pk = registerPassKey.publicKey; // obtained from the registration response
+
+        // Sign Passkey Public Key using Account Key, this is used to prove ownership of the account
+        const account_sig_passkey_pk = account_keypair.sign(passkey_pk);
+      ```
```
Comment (JoeCap08055) [inline designdocs/passkey_p256.md:200]: Perhaps either:
(a) interpolate code block in the description of each step, ie:
1.  **User visits Wallet Setup Page**
(code block)

2. etc...
(code block)


---or---
keep the entire code block where it is, but number the comments in the block to match the descriptions above

File: designdocs/passkey_p256.md [hunk 14]
```diff
@@ -0,0 +1,556 @@
...
+      // The signature generated by Passkey
+      passkey_sig_tx = passkeyTxSignature.signature;
+      passkey_authenticator_data = passkeyTxSignature.authenticatorData;
+      passkey_client_data = passkeyTxSignature.clientData;
+      //
+    ```
+
+3. **Transaction Submission**:
+   - The transaction payload, that requires, `passkey_pk`, `passkey_sig_tx`, `passkey_authenticator_data`, `passkey_client_data` is submitted to the chain.
+   - The transaction is submitted to the chain as in `unsigned` extrinsic.
```
Comment (JoeCap08055) [inline designdocs/passkey_p256.md:273]:
```suggestion
   - The transaction payload (consisting of `passkey_pk`, `passkey_sig_tx`, `passkey_authenticator_data`, `passkey_client_data`) is submitted to the chain.
   - The transaction is submitted to the chain as an `unsigned` extrinsic.
```

Comment (JoeCap08055) [reply designdocs/passkey_p256.md:273]: Also, might want a "see note" here for the "submitted as an unsigned extrinsic", and link to the discussion point later on about tradeoffs between unsigned vs. signed extrinsics

File: designdocs/passkey_p256.md [hunk 15]
```diff
@@ -0,0 +1,556 @@
...
+    ```
+
+4. **Backend Validation and Submission**:
+    - The backend can further validate the transaction payload before submitting it to the chain.
+    - The backend validates the `passkey_sig_tx` and the `passkey_authenticator_data` to ensure the authenticity of the transaction.
+    - For more secure operations, handling the transaction creation and submission on client side is recommended.
+
+### Security Considerations
+
+#### Generic
```
Comment (JoeCap08055) [inline designdocs/passkey_p256.md:300]:
```suggestion
#### General
```

File: designdocs/passkey_p256.md (hunk 12 above)
Comment (saraswatpuneet) [reply designdocs/passkey_p256.md]: Good suggestion, this helps graps the confusing aspect of many key names nicely

File: designdocs/passkey_p256.md [hunk 16]
```diff
@@ -0,0 +1,556 @@
...
+- Some possible optimizations
+  - **Compressed public key**: Currently passed public key is in **Cose** format, and it's between 70-73 bytes.
+      If the client can parse the Cose public key can extract the compressed encoded key. This public key can
+      get reduced to 33 bytes.
+  - **Challenge data deduplication**: Currently the challenge data is duplicated in `expected_challenge` and
+      in it's serialized format inside `passkey_client_data_json`. If the client is able to parse
+      `passkey_client_data_json` and replace `challenge` field value with empty string. Then during the
+      signature check we can replace that empty string with `expected_challenge` and that would allow us to
+      reduce the transaction size by around **40%**. It is important that the order of the field `passkey_client_data_json`
+      does not change during this operation since that would generate a different signature.
```
Comment (JoeCap08055) [inline designdocs/passkey_p256.md:419]:
```suggestion
  - **Challenge data deduplication**: Currently the challenge data is duplicated in `expected_challenge` and
      in it's serialized format inside `passkey_client_data_json`. If the client is able to parse
      `passkey_client_data_json` and replace `challenge` field value with empty string, then during the
      signature check we can replace that empty string with `expected_challenge` which would allow us to
      reduce the transaction size by around **40%**. It is important that the order of the field `passkey_client_data_json`  does not change during this operation since that would generate a different signature.
```

File: designdocs/passkey_p256.md [hunk 17]
```diff
@@ -0,0 +1,556 @@
...
+    }
+```
+
+## 7. Options for Discussion
+
+### Mortality of `account_sig_passkey_pk` i.e. Account Signature on PassKey Public Key
+
+#### Pros/Cons
+
+- **Pro**: Mortality of `account_sig_passkey_pk` would ensure that the account signature on the PassKey public key is only valid for a certain period, albieit it would be a complex setup.
```
Comment (JoeCap08055) [inline designdocs/passkey_p256.md]:
```suggestion
- **Pro**: Mortality of `account_sig_passkey_pk` would ensure that the account signature on the PassKey public key is only valid for a certain period
- **Con**: Complex to set up.
```

File: designdocs/passkey_p256.md [hunk 18]
```diff
@@ -0,0 +1,556 @@
...
+
+  >Wil wondered about a mortality as well, but I >have two fears:
```
Comment (JoeCap08055) [inline designdocs/passkey_p256.md:484]:
```suggestion
  >Wil wondered about a mortality as well, but I  have two fears:
```

File: designdocs/passkey_p256.md [hunk 19]
```diff
@@ -0,0 +1,556 @@
...
+As we know every signed extrinsic is required to get validated by a list of SignedExtensions which
+verify that extrinsic from different aspects. Depending on our choice to use _SignedExtrinsic_ or
+an _Unsigned_ one we would need to modify or re-implement some of the following.
+
+- **CheckNonce**: The nonce check should happen with the nonce inside the payload to avoid replay attacks.
+- **ChargeFrqTransactionPayment**: The transaction fee should be applied to the owner account inside
+the payload.
+- **CheckP256Signature**: A new extension should be written to verify the P256 signature inside the payload.
+- **Others**: We might also need to check the **Era**, **Mortality** and **Genesis hash** if we choose the
+unsigned path.
```
[the remark this answers is not in the document]
Comment (saraswatpuneet) [reply designdocs/passkey_p256.md:343]: Cleaner

File: designdocs/passkey_p256.md (hunk 14 above)
Comment (saraswatpuneet) [reply designdocs/passkey_p256.md]: Good suggestion, Added a link to discussion  :+1: 
> The transaction is submitted to the chain as an `unsigned` extrinsic.

File: designdocs/passkey_p256.md [hunk 20]
```diff
@@ -0,0 +1,556 @@
...
+- If signature checks implemented **on_validate** are expensive, then this would open a vulnerability
+  surface for DOS attacks.
+
+## 6. Implementation
+
+### On-chain Implementation
+
+#### Extrinsic verification
+
+Currently, the Frequency chain primarily makes use of _SignedExtrinsic_. As we know, every signed extrinsic is validated by a list of SignedExtensions which each verify a different aspect of that extrinsic. The list of signed extensions includes:
```
Comment (mattheworris) [inline designdocs/passkey_p256.md:334]: How much do we need to explain about Polkadot/Substrate? Should there be pointers to substrate docs?

```suggestion
Currently, the Frequency chain primarily makes use of _SignedExtrinsic_. Every signed extrinsic is validated by a list of SignedExtensions which each verify a different aspect of that extrinsic. The list of signed extensions includes:
```

File: designdocs/passkey_p256.md [hunk 21]
```diff
@@ -0,0 +1,556 @@
...
+  // finally verify the PassKey signature against the payload
+  verify_key
+   .verify(&passkey_signature_payload, &passkey_signature)
+ }
+```
+
+- Some possible optimizations
+  - **Compressed Public Key**: Currently passed Public Key is in **Cose** format, and it's between 70-73 bytes.
+      If the client can parse the Cose Public Key can extract the compressed encoded key. This Public Key can
+      get reduced to 33 bytes.
```
Comment (mattheworris) [inline designdocs/passkey_p256.md:414]:
```suggestion
If the client is capable of parsing the Cose Public Key, it can extract the compressed encoded key. This process can reduce the size of the Public Key to just 33 bytes.
```

File: designdocs/passkey_p256.md [hunk 22]
```diff
@@ -0,0 +1,556 @@
...
+  - **Challenge data deduplication**: Currently the challenge data is duplicated in `expected_challenge` and
+      in it's serialized format inside `passkey_client_data_json`. If the client is able to parse
+      `passkey_client_data_json` and replace `challenge` field value with empty string, then during the
+      signature check we can replace that empty string with `expected_challenge` which would allow us to
+      reduce the transaction size by around **40%**. It is important that the order of the field `passkey_client_data_json`  does not change during this operation since that would generate a different signature.
```
Comment (mattheworris) [inline designdocs/passkey_p256.md:419]:
```suggestion
  - **Challenge data deduplication**: Currently the challenge data is duplicated in `expected_challenge` and
      in it's serialized format inside `passkey_client_data_json`. If the client is able to parse
      `passkey_client_data_json` and replace the `challenge` field value with an empty string, then during the
      signature check we can replace that empty string with `expected_challenge` which would allow us to
      reduce the transaction size by around **40%**. It is important that the order of the field `passkey_client_data_json`  does not change during this operation since that would generate a different signature.
```

File: designdocs/passkey_p256.md [hunk 23]
```diff
@@ -0,0 +1,556 @@
...
+  - **Signing hash of payload**: Since there might be a size limit for the data that can get signed
+      via passkeys, it would make sense to only sign the cryptographic hash of the payload. This would also
+      reduce the challenge size to only 32 bytes if we decided not to implement the deduplication option.
+
+### Backend Example Code
+
+#### PassKey Registration Verification
+
+```typescript
+    // RegistrationResponseJSON is the data that PassKey authenticator returns after registeration
```
Comment (mattheworris) [inline designdocs/passkey_p256.md:429]:
```suggestion
    // RegistrationResponseJSON is the data that PassKey authenticator returns after registration
```

File: designdocs/passkey_p256.md [hunk 24]
```diff
@@ -0,0 +1,556 @@
...
+    const verification = await verifyRegistrationResponse({
+        response: body as RegistrationResponseJSON,
+        // challenge that server expects and provided for registeration
```
Comment (mattheworris) [inline designdocs/passkey_p256.md:432]:
```suggestion
        // challenge that server expects and provided for registration
```

**Review by mattheworris — APPROVED** [review]
- Read through doc
- Non-blocking comments (mostly typos)

File: designdocs/passkey_p256.md [hunk 25]
```diff
@@ -0,0 +1,556 @@
...
+- **Interoperability**: The use of standard cryptographic protocols (e.g., P256) ensures compatibility with other systems and services that also support these standards. This can facilitate integrations with existing wallets, security modules, and third-party applications.
+
+- **Backup and Recovery**: PassKey support can be used to facilitate account recovery and backup mechanisms. By signing the account Public Key, users can prove ownership of the account and recover access in case of a lost seed.
+
+## 3. Terminology
+
+### Accounts
+
+- **PassKey Account**: The user account associated with the PassKey.
+- **Frequency Account**: The account used for on-chain transactions and interactions. This account is derived from a seed phrase generated by Frequency Access.
```
Comment (wilwade) [inline designdocs/passkey_p256.md:65]:
```suggestion
- **Frequency Account**: The account used for on-chain transactions and interactions. This account is derived from a seed phrase generated by the Web Wallet.
```

File: designdocs/passkey_p256.md [hunk 26]
```diff
@@ -0,0 +1,556 @@
+# Design Doc for P256 PassKey Support
```
Comment (wilwade) [inline designdocs/passkey_p256.md:1]: Looks like the standard is with a lower case K. `/s/assKey/asskey/g`

File: designdocs/passkey_p256.md [hunk 27]
```diff
@@ -0,0 +1,556 @@
...
+      passkeyPublicKey: `passkey_pk`;
+      passkeySignature: `passkey_sig_tx`;
+      passkeyAuthenticator: `passkey_authenticator_data`;
+      passkeyClientDataJson: `passkey_client_data`;
+      passkeyCall: PasskeyCall;
+    }
+
+    interface PasskeyCall {
+      accountId: `account_pk`;
+      accountOwnershipProof: `AccountOwnershipProof`;
```
Comment (wilwade) [inline designdocs/passkey_p256.md:145]: Aligning with the other types without the backticks
```suggestion
      accountOwnershipProof: AccountOwnershipProof;
```

File: designdocs/passkey_p256.md [hunk 28]
```diff
@@ -0,0 +1,556 @@
...
+## 5. Specification
+
+### Terms
+
+These are the following technical terms and specifications for the implementation of PassKey support on Frequency chain.
+
+- `account_keypair` : A cryptographic key pair generated from a seed phrase
+- `account_pk` : Account Public Key
+- `passkey_pk` : PassKey Public Key
+- `passkey_sig_account_pk` : Signature generated by PassKey on  `account_pk`
```
Comment (wilwade) [inline designdocs/passkey_p256.md:122]: Might be worth making it clear this one is just for the web wallet backend?
```suggestion
- `passkey_sig_account_pk` : Signature generated by PassKey on  `account_pk` for Web Wallet backend verification
```

File: designdocs/passkey_p256.md (hunk 25 above)
Comment (saraswatpuneet) [reply designdocs/passkey_p256.md:65]: Nice 👍🏽

File: designdocs/passkey_p256.md (hunk 26 above)
Comment (saraswatpuneet) [reply designdocs/passkey_p256.md:1]: I will change it to `Passkey`

**Review by wilwade — APPROVED** [review]
Looks good to me.

I think all the remaining questions can/should be resolved in the implementation.

**Review by JoeCap08055 — APPROVED** [review]

File: designdocs/passkey_p256.md [hunk 29]
```diff
@@ -0,0 +1,555 @@
...
+### Signatures
+
+- **Account signature on Passkey Public Key**: A cryptographic signature generated using the Account Key. The data being signed is the Passkey Public Key. This is used to prove ownership of Account Key.
+- **Passkey Signatures**: A cryptographic signature generated using the Passkey private key. This is presented to Passkey enabled services as a challenge-response mechanism. Passkeys are used to generate two signatures as follows:
+  - **Signature on Account Public Key**: Passkey signs a message containing the account Public Key. This signature is retained by a Web Wallet platform and/or maybe used for account recovery.
+  - **Signature on Transactions**: Passkey signs the transaction payload which needs to be submitted on-chain. This signature is used to verify the authenticity of the transaction.
+
+## 4.1. Data Flow Diagram
+
+- Simplified Registration Diagram
```
Comment (shannonwells) [inline designdocs/passkey_p256.md:84]: These diagram links are broken for me, even when signed into my Google account.

**Review by shannonwells — COMMENTED** [review]
The image links need to be fixed.

File: designdocs/passkey_p256.md (hunk 29 above)
Comment (wilwade) [reply designdocs/passkey_p256.md:84]: Odd. They worked for me when they switched, but not anymore.

@aramikm if they were in drawio this might help: https://www.drawio.com/blog/embed-diagrams-github-markdown

**Comment by aramikm** [comment]
@shannonwells  It looks like github is changing URL's under the hood for uploaded data you can access the diagrams in https://github.com/frequency-chain/frequency/issues/1999#issuecomment-2159003597 until we find another alternative to uploading as content.

File: designdocs/passkey_p256.md (hunk 29 above)
Comment (aramikm) [reply designdocs/passkey_p256.md:84]: I'll take a look

File: designdocs/passkey_p256.md [hunk 30]
```diff
@@ -0,0 +1,555 @@
...
+        - [Pros/Cons](#proscons-1)
+      - [Extending MultiSignature](#extending-multisignature)
+        - [Pros/Cons](#proscons-2)
+    - [Generic Key Support](#generic-key-support)
+    - [Separate Pallet](#separate-pallet)
+  - [8. References](#8-references)
+
+## 1. Introduction
+
+This document outlines the design considerations and specifications for integrating P256 Passkey support for performing transactions on the Frequency chain. Passkey support aims to provide a novel non-custodial solution for managing user accounts and signing transactions on-chain.
```
Comment (enddynayn) [inline designdocs/passkey_p256.md:48]: Are these all transactions?

File: designdocs/passkey_p256.md [hunk 31]
```diff
@@ -0,0 +1,555 @@
...
+- **Interoperability**: The use of standard cryptographic protocols (e.g., P256) ensures compatibility with other systems and services that also support these standards. This can facilitate integrations with existing wallets, security modules, and third-party applications.
+
+- **Backup and Recovery**: Passkey support can be used to facilitate account recovery and backup mechanisms. By signing the account Public Key, users can prove ownership of the account and recover access in case of a lost seed.
+
+## 3. Terminology
+
+### Accounts
+
+- **Passkey Account**: The user account associated with the Passkey.
+- **Frequency Account**: The account used for on-chain transactions and interactions. This account is derived from a seed phrase generated by the Web Wallet.
```
Comment (enddynayn) [inline designdocs/passkey_p256.md:65]: What is the relationship of Frequency Account to an MSA account?

File: designdocs/passkey_p256.md [hunk 32]
```diff
@@ -0,0 +1,555 @@
...
+
+## 2. Benefits
+
+- **Non-Custodial Account Management**: Passkey support enables users to interact with the Frequency chain without actual seed access. This reduces the risk of seed exposure and enhances security.
+
+- **Functionality**: Passkey support provides a better user experience by enabling users to sign transactions without seed access. The proliferation of Passkey-enabled services can lead to a seamless user experience.
+
+- **Interoperability**: The use of standard cryptographic protocols (e.g., P256) ensures compatibility with other systems and services that also support these standards. This can facilitate integrations with existing wallets, security modules, and third-party applications.
+
+- **Backup and Recovery**: Passkey support can be used to facilitate account recovery and backup mechanisms. By signing the account Public Key, users can prove ownership of the account and recover access in case of a lost seed.
```
Comment (enddynayn) [inline designdocs/passkey_p256.md:58]: I thought passkeys are seedless?

**Review by shannonwells — APPROVED** [review]
Good to go, all my feedback was addressed.
